astropy/timeseries/periodograms/lombscargle/implementations/main.py

```diff
@@ -218,6 +218,10 @@ def lombscargle(
         )
         kwds.update(f0=f0, df=df, Nf=Nf)
 
+    # enable optimizations specific to regular grids only
+    if method == "cython":
```

Comment:
This new option needs to be documented in the docstring, at least.

Reply:
the *assumptions* for this methods (e.g., requiring a regular grid) should be documented as well